### ISSUE TITLE
Unexpected synonym behavior

### ISSUE THREAD

**Issue by bidoubiwa** [meilisearch/MeiliSearch#763]
**Describe the bug**

When I add the following synonym `"DBZ" : ["Dragon Ball"]` I expect that when I search for `DBZ` i will also receive documents containing `Dragon Ball` (of course after the exact matches with DBZ). 
This is not what is happening, as I feel like the synonym is completely ignored.

I know that when I started adding synonyms they were taking into account. Once I made more and more changes on the settings, somewhere synonyms were ignored. 

I added the synonyms again, to make them work again, but it did not work. There are still ignored.

If I recall those were my changes 
- Changed synonyms in settings route
- Changed synonyms in sub route
- Changed searchable Attributes (via sub route)
- Changed settings in settings route (again searchable attributes where changed)
- Changed synonyms

When searching for DBZ the only result I get is this: 

<img width="973" alt="Screenshot 2020-06-08 at 17 04 33" src="https://user-images.githubusercontent.com/33010418/84046577-9191b000-a9aa-11ea-8c7b-cbc82d10b874.png">

**To Reproduce**
1. Add 200mb movie dataset
2. Update settings multiple times until it looks like this
```json
{
  "rankingRules": [
    "typo",
    "words",
    "proximity",
    "attribute",
    "desc(vote_count)",
    "wordsPosition",
    "exactness",
    "desc(release_date)",
    "desc(vote_average)"
  ],
  "distinctAttribute": null,
  "searchableAttributes": [
    "vote_count",
    "release_date",
    "vote_average",
    "id",
    "title",
    "tagline",
    "overview",
    "cast",
    "popularity"
  ],
  "displayedAttributes": [
    "overview",
    "tagline",
    "cast",
    "release_date",
    "vote_average",
    "title",
    "poster_path",
    "vote_count",
    "id",
    "popularity"
  ],
  "stopWords": [],
  "synonyms": {
    "DBZ": [
      "Dragon Ball"
    ],
    "HP": [
      "Harry Potter"
    ]
  },
  "acceptNewFields": true,
  "attributesForFaceting": [
    "director",
    "producer",
    "genres",
    "production_companies"
  ]
}
```
2. Search for `DBZ`

**Expected behavior**
Having movies with `Dragon Ball` 

**Screenshots**
I want results from both screens
<img width="973" alt="Screenshot 2020-06-08 at 17 04 33" src="https://user-images.githubusercontent.com/33010418/84046452-6b6c1000-a9aa-11ea-8201-7f62ff06daf3.png">

<img width="953" alt="Screenshot 2020-06-08 at 17 06 07" src="https://user-images.githubusercontent.com/33010418/84046433-66a75c00-a9aa-11ea-96a8-0641f47805ce.png">

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] MAC catalina
 - Browser [e.g. chrome, safari] Chrome
 - Version [e.g. 22] master

**Comment by MarinPostma** [comment]
possible duplicate of #656

**Comment by MarinPostma** [comment]
I have investigated this isssue. From what I have collected, this is due to the case sensitivity of the synonyms. This is part of the [tokenizer rfc](https://github.com/meilisearch/specifications/pull/2). This issue is also reported in #964  and #1048 

The way to deal with that is to put all synonyms in lowercase:
```
"synonyms": {
    "dbz": [
      "dragon ball", "Dragon Ball"
    ],
``` 
this is not ideal, and will be addressed with the new version of the tokenizer.
